fixtures/trait-methods/src/lib.rs

```diff
@@ -267,4 +249,253 @@ impl Hash for UdlEnum {
     }
 }
 
+// flat enum with Display only - Kotlin doesn't support Eq/Ord/Hash exports for flat enums
+#[derive(uniffi::Enum, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[uniffi::export(Debug, Display)]
+pub enum FlatTraitEnum {
+    Alpha,
+    Beta,
+    Gamma,
+}
+
+impl std::fmt::Display for FlatTraitEnum {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let label = match self {
+            FlatTraitEnum::Alpha => "flat-alpha",
+            FlatTraitEnum::Beta => "flat-beta",
+            FlatTraitEnum::Gamma => "flat-gamma",
+        };
+        write!(f, "FlatTraitEnum::{label}")
+    }
+}
+
+#[uniffi::export]
+fn get_flat_trait_enum(index: u8) -> FlatTraitEnum {
+    match index {
+        0 => FlatTraitEnum::Alpha,
+        1 => FlatTraitEnum::Beta,
+        _ => FlatTraitEnum::Gamma,
+    }
+}
+
+// flat error - variants have no fields
+#[derive(Debug, uniffi::Error, thiserror::Error, Clone, PartialEq, Eq, Hash)]
+#[uniffi::export(Display)]
+pub enum FlatError {
+    #[error("error: not found")]
+    NotFound,
+    #[error("error: unauthorized")]
+    Unauthorized,
+    #[error("error: internal error")]
+    InternalError,
+}
+
+#[uniffi::export]
+fn throw_flat_error(i: u8) -> Result<(), FlatError> {
+    match i {
+        0 => Err(FlatError::NotFound),
+        1 => Err(FlatError::Unauthorized),
+        _ => Err(FlatError::InternalError),
+    }
+}
+
+// mixed error with all possible exported traits
```

Comment:
I don't think we need such rich error tests - can we get away with 1 error type and a few variants? eg, we don't need the "error that doesn't end in "Error"" complication.